[igezt] iP

igezt opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind from having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use
- All you need to do is:

  1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
  2. double-click it.
  3. add your tasks.
  4. let it manage your tasks for you 😉


And it is **FREE**!

Features:

 

- [x] Managing tasks
- [x]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main`  method:

``` java
public class Main {
   public static void main(String[] args) {
       Application.launch(MainApp.class, args);
 }
}
```

Commit:
Add Gradle support

igezt committed:
Level-1

igezt committed:
Level-2

igezt committed:
Level-3

igezt committed:
Level-4

igezt committed:
A-TextUiTesting

igezt committed:
Level-5

igezt committed:
Level-6

igezt committed:
A-TextUiTesting

igezt committed:
Level-7

igezt committed:
Level-8

igezt committed:
Merge branch 'branch-Level-7' into branch-level-8

# Conflicts:
#	src/main/java/DukeBot/Deadline.java
#	src/main/java/DukeBot/DukeBot.java
#	src/main/java/DukeBot/Event.java

igezt committed:
Amend merge conflict for Level-7 and Level-8

igezt committed:
Amend imports

shaowi commented on src/main/java/DukeBot/DukeBot.java:
```diff
@@ -0,0 +1,288 @@
+
+        // Extract deadline date and task item.
+        String[] lines = parameters.split(" ");
+        boolean by = false;
```
Can name it as hasByDatetime or hasByDT to be more explicit? :)

shaowi commented on src/main/java/DukeBot/DukeBot.java:
```diff
@@ -0,0 +1,288 @@
+        this.dataBase.update(this.list);
+        return FRAME +
+                "     Bye. Hope to see you again soon!\n" +
+                FRAME;
```
Instead of writing return FRAME ... on multiple sections? you may extract it to a method and pass in the required message output as parameter instead?

shaowi commented on src/main/java/DukeBot/ToDo.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Override
+    public ArrayList<String> data() {
+        ArrayList<String> data = new ArrayList<>();
```
Plural form of data recommended and a more explicit name can be taskDatas?

igezt committed:
A-MoreOOP

igezt committed:
A-Packages

igezt committed:
Changed names of package

igezt committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
A-Gradle

igezt committed:
A-Gradle

igezt committed:
A-JUnit
Changed the responses for adding a Deadline, Event and Todos

igezt committed:
A-JavaDoc

igezt committed:
A-CodingStandard

igezt committed:
Level-9

igezt committed:
Merge branch 'branch-A-CodingStandard'

igezt committed:
Merge branch 'branch-A-JavaDoc'
Resolved Conflicts

igezt committed:
A-CheckStyle added

zuohui48 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,115 @@
+package duke.parser;
+
+import duke.commands.*;
```
Are we allowed to use wildcard imports for methods of our own classes?

zuohui48 commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,53 @@
+     * Returns the status of the ToDo task.
+     */
+    @Override
+    public String status() {
```
Will getStatus() be a better name for this method?

zuohui48 commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,53 @@
+     * @return Returns all relevant information of the task in an ArrayList of Strings to be saved into the Database.
+     */
+    @Override
+    public ArrayList<String> data() {
```
Will getData() be a better name for this method?

zuohui48 commented on src/main/java/duke/database/Database.java:
```diff
@@ -0,0 +1,94 @@
+/** Represents the database storage for Duke using a basic text file. */
+public class Database {
+
+    //ASCII 31 is known as the "Unit Seperator"
+    //Value taken from
+    //https://stackoverflow.com/questions/492090/least-used-delimiter-character-in-normal-text-ascii-128
```
These comments really helped me understand what the following lines of codes are doing!

zuohui48 reviewed (commented):
Overall, I think your codes are nicely written and mostly follow the coding standard.

shittake commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * Represents the abstract Command to be inherited from by all Commands.
+ */
+public abstract class Command {
```
I like how you created an abstract class with a static final String `FRAME` that all other Commands could inherit from

shittake commented on src/main/java/duke/database/Database.java:
```diff
@@ -0,0 +1,94 @@
+                    break;
+                case "D":
+                    tasks.add(new Deadline(taskData));
+                }
```
I like your use of switch cases to indicate the type of Task to be added, which makes it clean to read. Also, you did not indent the `case` statements, which is commendable as many people (including myself) tend to forget about it.

shittake commented on src/main/java/duke/exception/blankfieldexceptions/BlankFieldDeadlineException.java:
```diff
@@ -0,0 +1,9 @@
+    public BlankFieldDeadlineException() {
+        super("\n" + "    ____________________________________________________________\n" + "\n" +
+                "     ☹ OOPS!!! The description or date of a deadline cannot be empty." + "\n" +
+                "    ____________________________________________________________\n");
```
I believe it would be better if you had not hardcoded the filler lines, and instead used a `final` String variable to store it, which will apply to all your Duke Exception subclasses.

shittake commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+        data.add(this.details);
+        data.add(this.deadline.toString());
+        return data;
+    }
```
I do not think it is a good idea to have a method name called `data()` which also collides with the name of your arraylist variable in the method. Perhaps a better method name would be `getFormattedData`

shittake commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,41 @@
+    /** Marks the task as incomplete */
+    public void incomplete() {
+        this.isCompleted = false;
+    }
```
I think if you are changing the states of the Task, better variable names would be `makeComplete()` and `makeIncomplete()` respectively

shittake commented on src/main/java/duke/commands/AddToDoCommand.java:
```diff
@@ -0,0 +1,43 @@
+        ui.response(FRAME + "\n" +
+                "     Got it. I've added this task:" + "\n" +
+                "     " + newToDo.status() + "\n" +
+                "     Now you have " + taskList.length() + " tasks in the list" + "\n" +
```
I notice you tend to have many of such hardcoded whitespaces in your printed output (including in other files). Perhaps it would be better to have a `final` String variable storing these whitespaces, or a `final int` storing the width of such whitespaces, for easy modification in case future changes are needed.

shittake reviewed (commented):
Looks good to merge!

Great job 👍

igezt committed:
Added a UI using JavaFX Level-10

shaowi reviewed (commented):
Looks good to merge!

igezt committed:
Edit for checkstyle

igezt committed:
Edit for Checkstyle

igezt committed:
Add assertions to all commands to check for correct isActive value.

igezt committed:
Change method names to comply with coding standards

igezt committed:
Parser: Shorten method body of parse by abstracting out parser logic
AddEventCommand: Use enums instead of an integer to represent the state of the parsing logic

igezt committed:
Parser: Shorten method body of parse by abstracting out parser logic
AddEventCommand: Use enums instead of an integer to represent the state of the parsing logic

igezt committed:
Remove trailing whitespaces for messages for both Duke and User messages
Has trailing whitespaces for all messages

Needs to change due to the extra space being taken up in the dialog box

Removed all trailing whitespaces to reduce space taken up in dialog box

So that more space can be given for the messages

igezt committed:
Move AddEventParserLogic into its own enum file

Initially was in the same file as other command classes

Could have caused confusion for other developers as it was not a command

Moved it to a separate file

To make sure that its easy to differentiate from the other files

igezt committed:
Change color of background of UI and made attempts to clean up css file
Edit ParserTest to reflect new DukeBot logic

igezt committed:
Merge pull request #3 from igezt/A-CodeQuality

A-code-quality

igezt committed:
Merge branch 'master' into A-Assertions

igezt committed:
A-Assertions Merge pull request #2 from igezt/A-Assertions

Add assertions to all commands to check for correct isActive value.

igezt committed:
Change UI to have the bot look better

igezt committed:
Adjust UI to make it size adjustable. A-BetterGui

igezt committed:
Add ability to edit tasks.

igezt committed:
Add ability to edit tasks and fix UI bugs

igezt committed:
Add more error handling especially for edit command related. Remove
task number out of loop error handling as it has already been checked in
a previous function

igezt committed:
Edit exception messages and fix UI messages with wrong indents.

igezt committed:
Update Javadocs to reflect coding standard

igezt committed:
Remove placeholder image for user. Add Ui.png

igezt committed:
Replace user image to not be placeholder.

igezt committed:
Update list command to account for if the task list is empty.

igezt committed:
Create user guide

igezt committed:
Edit user guide

igezt committed:
Edit changes for user guide

igezt committed:
Edit readme.md.

igezt committed:
Edit ReadMe grammer
